aws/table_aws_ssm_document.go

```diff
@@ -0,0 +1,262 @@
+		region = matrixRegion.(string)
+	}
+	plugin.Logger(ctx).Trace("listAwsSSMDocuments", "AWS_REGION", region)
+
+	// Create session
+	svc, err := SsmService(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	// List call
+	err = svc.ListDocumentsPages(
+		&ssm.ListDocumentsInput{},
+		func(page *ssm.ListDocumentsOutput, isLast bool) bool {
+			for _, documentIdentifier := range page.DocumentIdentifiers {
+				d.StreamListItem(ctx, &ssm.DocumentDescription{
+					Name:            documentIdentifier.Name,
+					TargetType:      documentIdentifier.TargetType,
+					DocumentVersion: documentIdentifier.DocumentVersion,
+					Owner:           documentIdentifier.Owner,
+					VersionName:     documentIdentifier.VersionName,
+					DocumentFormat:  documentIdentifier.DocumentFormat,
+					PlatformTypes:   documentIdentifier.PlatformTypes,
+					ReviewStatus:    documentIdentifier.ReviewStatus,
+					Tags:            documentIdentifier.Tags,
+					DocumentType:    documentIdentifier.DocumentType,
+				})
+
+			}
+			return !isLast
+		},
+	)
+
+	return nil, err
+}
+
+//// HYDRATE FUNCTIONS
+
+func getAwsSSMDocument(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	logger.Trace("getAwsSSMDocument")
+
+	// TODO put me in helper function
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+	var name string
+	if h.Item != nil {
+		name = *h.Item.(*ssm.DocumentDescription).Name
+	} else {
+		name = d.KeyColumnQuals["name"].GetStringValue()
+	}
+
+	// Create Session
+	svc, err := SsmService(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	// Build the params
+	params := &ssm.DescribeDocumentInput{
+		Name: &name,
+	}
+
+	// Get call
+	data, err := svc.DescribeDocument(params)
+	if err != nil {
+		logger.Debug("getAwsSSMDocument", "ERROR", err)
+		return nil, err
+	}
+
+	return data.Document, nil
+}
+
+func getAwsSSMDocumentAkas(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getAwsSSMDocumentAkas")
+	name := *h.Item.(*ssm.DocumentDescription).Name
+	c, err := getCommonColumns(ctx, d, h)
+	if err != nil {
+		return nil, err
+	}
+	commonColumnData := c.(*awsCommonColumnData)
+	aka := "arn:" + commonColumnData.Partition + ":ssm:" + commonColumnData.Region + ":" + commonColumnData.AccountId + ":document"
+
+	if strings.HasPrefix(name, "/") {
+		aka = aka + name
+	} else {
+		aka = aka + "/" + name
+	}
+
+	return []string{aka}, nil
+}
+
+func ssmDocumentTagListToTurbotTags(ctx context.Context, d *transform.TransformData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("ssmDocumentTagListToTurbotTags")
+	document := d.HydrateItem.(*ssm.DocumentDescription)
+
+	// Mapping the resource tags inside turbotTags
+	var turbotTagsMap map[string]string
+	if document != nil {
+		turbotTagsMap = map[string]string{}
```

Comment:
Return nil if the `Tags` value is null
```suggestion
		if document.Tags != nil {
		    return nil, nil
		}
		turbotTagsMap = map[string]string{}
```

Reply:
@anisadas Was this change made?